controlplane/kubeadm/internal/controllers/scale_test.go

```diff
@@ -32,22 +35,66 @@ import (
 	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
 	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
 	"sigs.k8s.io/cluster-api/controlplane/kubeadm/internal"
+	"sigs.k8s.io/cluster-api/util"
 	"sigs.k8s.io/cluster-api/util/collections"
 	"sigs.k8s.io/cluster-api/util/conditions"
 )
 
 func TestKubeadmControlPlaneReconciler_initializeControlPlane(t *testing.T) {
-	g := NewWithT(t)
+	setup := func(t *testing.T, g *WithT) *corev1.Namespace {
+		t.Helper()
+
+		t.Log("Creating the namespace")
+		ns, err := env.CreateNamespace(ctx, "test-machine-set-reconciler-sync-machines")
+		g.Expect(err).To(BeNil())
+
+		return ns
+	}
+
+	teardown := func(t *testing.T, g *WithT, ns *corev1.Namespace) {
+		t.Helper()
 
-	cluster, kcp, genericMachineTemplate := createClusterWithControlPlane(metav1.NamespaceDefault)
+		t.Log("Deleting the namespace")
+		g.Expect(env.Delete(ctx, ns)).To(Succeed())
+	}
 
-	fakeClient := newFakeClient(cluster.DeepCopy(), kcp.DeepCopy(), genericMachineTemplate.DeepCopy())
+	g := NewWithT(t)
+	namespace := setup(t, g)
+	defer teardown(t, g, namespace)
+
+	cluster, kcp, _ := createClusterWithControlPlane(namespace.Name)
```

Comment:
Took another look. Can you please check how much effort it would be to just already modify the `createClusterWithControlPlane` func to the new type?

I think it could just work even in the other cases and then we can avoid a lot of new code (defining the new object everywhere + modifying the infra ref)

Reply:
Checked it and luckily refactoring this to use the InfraMachineTemplate from the builder was simple enough. Switched it.

Reply:
Nice! Less noise on this PR and less tech debt :)